protocol-designer/src/pages/StartingDeckState/DeckSetupTools.tsx

```diff
@@ -0,0 +1,330 @@
+    ModuleModel | Fixture | null
+  >(createdModuleForSlot?.model ?? preSelectedFixture ?? null)
+
+  const [selecteLabwareDefURI, setSelectedLabwareDefURI] = React.useState<
+    string | null
+  >(createdLabwareForSlot?.labwareDefURI ?? null)
+  const [
+    nestedSelectedLabwareDefURI,
+    setNestedSelectedLabwareDefURI,
+  ] = React.useState<string | null>(
+    createdNestedLabwareForSlot?.labwareDefURI ?? null
+  )
+  const moduleModels = getModuleModelsBySlot(
+    enableAbsorbanceReader,
+    robotType,
+    slot
+  )
+  const [tab, setTab] = React.useState<'hardware' | 'labware'>(
+    moduleModels.length === 0 ? 'labware' : 'hardware'
+  )
+
+  let fixtures: Fixture[] = []
+  if (slot === 'D3') {
+    fixtures = FIXTURES
+  } else if (['A3', 'B3', 'C3'].includes(slot)) {
+    fixtures = ['stagingArea', 'trashBin']
+  } else if (['A1', 'B1', 'C1', 'D1'].includes(slot)) {
+    fixtures = ['trashBin']
+  }
+
+  const hardwareTab = {
+    text: t('deck_hardware'),
+    disabled: moduleModels.length === 0,
+    isActive: tab === 'hardware',
+    onClick: () => {
+      setTab('hardware')
+    },
+  }
+  const labwareTab = {
+    text: t('labware'),
+    disabled:
+      selectedHardware === 'wasteChute' ||
+      selectedHardware === 'wasteChuteAndStagingArea' ||
+      selectedHardware === 'trashBin',
+    isActive: tab === 'labware',
+    onClick: () => {
+      setTab('labware')
+    },
+  }
+
+  const handleClear = (): void => {
+    //  clear module from slot
+    if (createdModuleForSlot != null) {
+      dispatch(deleteModule(createdModuleForSlot.id))
+    }
+    //  clear fixture(s) from slot
+    if (createFixtureForSlots.length > 0) {
+      createFixtureForSlots.forEach(fixture =>
+        dispatch(deleteDeckFixture(fixture.id))
+      )
+    }
+    //  clear labware from slot
+    if (createdLabwareForSlot != null) {
+      dispatch(deleteContainer({ labwareId: createdLabwareForSlot.id }))
+    }
+    //  clear nested labware from slot
+    if (createdNestedLabwareForSlot != null) {
+      dispatch(deleteContainer({ labwareId: createdNestedLabwareForSlot.id }))
+    }
+  }
+
+  const handleConfirm = (): void => {
+    //  clear entities first before recreating them
+    handleClear()
+    const fixture = FIXTURES.includes(selectedHardware as Fixture)
+      ? (selectedHardware as DeckFixture | 'wasteChuteAndStagingArea')
+      : undefined
+    const moduleModel = MODULE_MODELS.includes(selectedHardware as ModuleModel)
+      ? (selectedHardware as ModuleModel)
+      : undefined
+
+    if (fixture != null) {
+      //  create fixture(s)
+      if (fixture === 'wasteChuteAndStagingArea') {
+        dispatch(createDeckFixture('wasteChute', cutoutId))
+        dispatch(createDeckFixture('stagingArea', cutoutId))
+      } else {
+        dispatch(createDeckFixture(fixture, cutoutId))
+      }
+    }
+    if (moduleModel != null) {
+      //  create module
+      dispatch(
+        createModule({
+          slot: slot,
```

Comment:
to align with the L195 and L211
```suggestion
          slot,
```